### PR TITLE
Invite component

### LINKED ISSUE #8: Create Invite Component
The component should have multiple states: 

1. Previously opened
2. On hover: Envelope appears to be slightly open
3. Unopened (has the blue dot in the corner)

Invite

### PR DESCRIPTION
**Pull request by josiahsrc**
This PR:
- Adds support for the invitation component which has a nifty animation flap thing. The blue notification dot can be toggled on or off via the `isUnopened` param.
- Introduces a reusable component for the home page view cards called `ActionCard`. This component extends the already existing card from the common components.
- ~~Adds in 2 color swatches (primary and secondary) which can be used to access the shades of the main app colors.~~

Screen-Recording-2020-10-09-at-11 26 14-PM

Close: #8

### CODE REVIEW

File: client/src/components/Common/Card/index.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import React from "react";
 import styled from "styled-components";
-import { spacing16 } from "../../../resources/style-constants";
 
 const Wrapper = styled.div`
   z-index: 3;
   box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
-  padding: ${spacing16};
```
Comment (josiahsrc) [inline client/src/components/Common/Card/index.js:8]: I removed this padding from here to make the card usable in different contexts. I fixed the usages before making this change.

File: client/src/components/Home/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { spacing64 } from "../../resources/style-constants";
+
+import Invitation from './Invitation';
+import ActionCard from './ActionCard';
+
+const Wrapper = styled.div`
+  padding: ${spacing64};
+`;
+
+const Home = () => {
```
Comment (josiahsrc) [inline client/src/components/Home/index.js]: I also added a home page component. I can remove this if you have other plans for the home page component.

File: client/src/resources/style-constants.js [hunk 3]
```diff
@@ -20,6 +20,20 @@ export const link = "#2196f3"; // color for links
 export const highlightedLink = "#35baf6"; // color of links on hover
 export const errorRed = "#FF0002";
 
+/*
+Swatches (shades of colors).
+*/
+export const primarySwatch = {
```
Comment (hokansom) [inline client/src/resources/style-constants.js]: We can switch to use these colors swatches if you want to define them all. I've been using the theme1 as the main color, and then theme2 is a slighter darker shade for the button hover colors. I'm fine either way, but I do want to be consistent throughout the entire project.

File: client/src/App.js [hunk 4]
```diff
@@ -13,6 +14,7 @@ class App extends Component {
         <Switch>
           <Route path="/" component={MainPage} exact />
           <Route path="/login" component={LoginPage} />
+          <Route path="/home" component={HomePage} />
```
Comment (hokansom) [inline client/src/App.js]: The home/main page will live at "/", but we can fix that once we actually assemble the page.

**Review by hokansom — CHANGES_REQUESTED** [review]
This looks great! So just two minor things. Since you removed the padding on the card, does the login page still look okay? We may need to add the padding on the login component directly. 

Second: if we are transitioning to using the color swatches, I would like the rest of the components to be updated to also use the theme color from the primarySwatch that way colors will be consistent throughout. (Then we can get rid of the theme1, theme2, etc.)

**Comment by josiahsrc** [comment]
@hokansom  Awesome, thanks for the review! 

For the padding on the card, the only way I could figure out how to generalize the `common/card` was to remove the padding. I left a comment about this, all usages of the common card have been updated to accommodate my change: https://github.com/Clotonervo/Event-Planner/pull/37#discussion_r502749484

As for the color swatches, what do you prefer? I threw the swatches out as an idea in this PR, but I'm happy revert to the theme1, theme2, etc 😄

File: client/src/App.js (hunk 4 above)
Comment (josiahsrc) [reply client/src/App.js]: Ah, missed that. Okay, I'll fix that up in this PR

File: client/src/resources/style-constants.js (hunk 3 above)
Comment (josiahsrc) [reply client/src/resources/style-constants.js]: Good point. I left a comment in the main thread about this. LMK

**Review by hokansom — APPROVED** [review]

**Pull request merged**